lib/rules/default-param-last.js

```diff
@@ -14,7 +27,8 @@ module.exports = {
             description: "Enforce default parameters to be last",
             recommended: false,
             frozen: true,
-            url: "https://eslint.org/docs/latest/rules/default-param-last"
+            url: "https://eslint.org/docs/latest/rules/default-param-last",
+            typescript: "syntax"
```

Comment:
> I'm also wondering if maybe we should add something into `meta` indicating that this rule is TypeScript-aware?

I found this to be a surprisingly difficult design question. The property needs to indicate:

* It _does_ support TypeScript _syntax_
* It _doesn't_ support TypeScript _types_

...and I couldn't figure out any nice succinct single-word property names for this. Using `typescript: "syntax"` here explicitly tells folks it's _only_ syntax, not types. It leaves an option for custom parsers like tseslint to later put in `"types"` too, maybe. But I'm not confident in this.